### PR TITLE
Upgrade ingress controller

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2024

Adds the changes suggested in https://github.com/giantswarm/giantswarm/issues/1875

### CODE REVIEW

File: v_0_1_0/master_template.go [hunk 1]
```diff
@@ -833,7 +833,8 @@ write_files:
       strategy:
         type: RollingUpdate
         rollingUpdate:
-          maxUnavailable: 1
+          maxSurge: 1
+          maxUnavailable: 0
```
Comment (calvix) [inline v_0_1_0/master_template.go]: what means maxUnavaiable 0? does that work?
we have nginx controller limited to 1 per machine  and there can be created another pod of we dont remove the previous

File: v_0_1_0/master_template.go [hunk 2]
```diff
@@ -880,6 +882,12 @@ write_files:
                 scheme: HTTP
               initialDelaySeconds: 10
               timeoutSeconds: 1
+            lifecycle:
+              preStop:
+                exec:
+                  command:
+                  - sleep
+                  - "15"
```
Comment (calvix) [inline v_0_1_0/master_template.go]: why do we wait 15 sec before deletion?

Comment (calvix) [reply v_0_1_0/master_template.go]: sorry read the issue and found the reason for this

File: v_0_1_0/master_template.go (hunk 1 above)
Comment (fgimenez) [reply v_0_1_0/master_template.go]: @calvix this comes too from the recommended settings in the linked issues, I've checked it and seems to work, deleted a ic pod, it moves to terminating and a new one is created while the previous one is still there, eventually the old one is gone and then the new one moves from pending to running, is that ok?

**Comment by fgimenez** [comment]
Tested on lycan, normal ingress, tls termination and ssl passthrough working.

File: v_0_1_0/master_template.go (hunk 1 above)
Comment (puja108) [reply v_0_1_0/master_template.go]: as we don't have hard-anti-affitini it should work

Comment (fgimenez) [reply v_0_1_0/master_template.go]: yep, also these settings make more sense imo than maxUnavailable=1 (given that only one runs on each host), they try make sure that there will be always (at most) one pending while any of them is terminating.

Comment (puja108) [reply v_0_1_0/master_template.go]: ack

Comment (calvix) [reply v_0_1_0/master_template.go]: we have hard affinity afaik

Comment (fgimenez) [reply v_0_1_0/master_template.go]: @calvix you are right, after trying a rolling update changing the image the new pod was getting stuck on pending with:
```
Events:
  FirstSeen    LastSeen    Count    From            SubObjectPath    Type        Reason            Message
  ---------    --------    -----    ----            -------------    --------    ------            -------
  3m        15s        15    default-scheduler            Warning        FailedScheduling    No nodes are available that match all of the predicates: PodFitsHostPorts (3), PodToleratesNodeTaints (1).
```
and the update didn't progress. With `maxUnavailable` set to 1 the rolling update works fine, already pushed this change.

**Pull request merged**